File: uploaddemo/src/business/galary/java/com/jw/galary/video/VideoItem.java
package com.jw.galary.video;

import android.os.Parcel;
import android.os.Parcelable;

import com.jw.galary.base.bean.BaseItem;

import java.io.Serializable;

public class VideoItem extends BaseItem implements Serializable, Parcelable {
    public String thumbPath;
    public long duration;
    public static final Creator<VideoItem> CREATOR = new Creator<VideoItem>() {
        public VideoItem createFromParcel(Parcel source) {
            return new VideoItem(source);
        }

        public VideoItem[] newArray(int size) {
            return new VideoItem[size];
        }
    };

    protected VideoItem(Parcel in) {
        setName(in.readString());
        setPath(in.readString());
        setSize(in.readLong());
        setMimeType(in.readString());
        duration = in.readLong();
        thumbPath = in.readString();
    }

    public int describeContents() {
        return 0;
    }

    public boolean equals(Object o) {
        if (o instanceof VideoItem) {
            VideoItem item = (VideoItem) o;
            return getPath().equalsIgnoreCase(item.getPath());
        } else {
            return super.equals(o);
        }
    }

    public VideoItem() {
    }

    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(getName());
        dest.writeString(getPath());
        dest.writeLong(getSize());
        dest.writeString(getMimeType());
        dest.writeLong(duration);
        dest.writeString(thumbPath);
    }
}

File: uploaddemo/src/business/galary/java/com/jw/galary/video/bean/VideoItem.kt
package com.jw.galary.video.bean

import android.os.Parcel
import android.os.Parcelable

import com.jw.galary.base.bean.BaseItem

import java.io.Serializable

class VideoItem : BaseItem, Serializable, Parcelable {
    lateinit var thumbPath: String
    var duration: Long = 0

    protected constructor(`in`: Parcel) {
        name = `in`.readString()
        path = `in`.readString()
        size = `in`.readLong()
        mimeType = `in`.readString()
        duration = `in`.readLong()
        thumbPath = `in`.readString()
    }

    override fun describeContents(): Int {
        return 0
    }

    override fun equals(o: Any?): Boolean {
        if (o is VideoItem) {
            val item = o as VideoItem?
            return path!!.equals(item!!.path!!, ignoreCase = true)
        } else {
            return super.equals(o)
        }
    }

    constructor() {}

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeString(name)
        dest.writeString(path)
        dest.writeLong(size)
        dest.writeString(mimeType)
        dest.writeLong(duration)
        dest.writeString(thumbPath)
    }

    companion object CREATOR : Parcelable.Creator<VideoItem> {
        override fun createFromParcel(parcel: Parcel): VideoItem {
            return VideoItem(parcel)
        }

        override fun newArray(size: Int): Array<VideoItem?> {
            return arrayOfNulls(size)
        }
    }
}
